terraform/modules/dandiset_bucket/main.tf

```diff
@@ -0,0 +1,79 @@
+resource "aws_s3_bucket" "dandiset_bucket" {
+  # provider = var.aws_provider
+
+  bucket = var.bucket_name
+  // Public access is granted via a bucket policy, not a canned ACL
+  acl = "private"
+
+  cors_rule {
+    allowed_origins = [
+      "*",
+    ]
+    allowed_methods = [
+      "PUT",
+      "POST",
+      "GET",
+      "DELETE",
+    ]
+    allowed_headers = [
+      "*"
+    ]
+    expose_headers = [
+      "ETag",
+    ]
+    max_age_seconds = 3000
+  }
+
+  logging {
+    target_bucket = aws_s3_bucket.log_bucket.id
+  }
+
+  versioning {
+    enabled = var.versioning
+  }
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
+resource "aws_s3_bucket_ownership_controls" "dandiset_bucket" {
+  bucket = aws_s3_bucket.dandiset_bucket.id
+
+  rule {
+    object_ownership = "BucketOwnerPreferred"
```

Comment:
I don't understand the details of how ownership is to be handled, but I would have expected this to be `BucketOwnerEnforced`.

Reply:
The sponsored bucket has `BucketOwnerPreferred`, but I'm not sure why. I assume that was the way it was given to us. I think `BucketOwnerEnforced` is the correct setting.

Reply:
`BucketOwnerEnforced` isn't an allowed string, I think you need to disable ACLs entirely. I'm going to save that work for later.